### PR TITLE
Enable connector templates per default

### LINKED ISSUE #4455: Simplify enabling of connectors within modeler
### What should we do?

Related to https://github.com/camunda/camunda-modeler/issues/4302 we want to ensure that it is simple to enable connectors within the app, i.e. on first time startup. This allows users to get a better out-of-the-box [experience for Camunda + connectors](https://github.com/camunda/product-hub/issues/2253#issuecomment-2277522319).

* [x] We explore introducing a minimal settings page to accomplish this, hence it is related to  https://github.com/camunda/camunda-modeler/issues/2913
  * [x] Not part of first iteration
* [x] We want to load/update connector templates lazily, only when a user opens a Camunda 8 diagram (https://github.com/camunda/camunda-modeler/pull/4474)
* [x] We want connectors to be enabled automatically, and indicate ability to configure via the what's new communication (https://github.com/camunda/camunda-modeler/pull/4474)
* [x] We want to update documentation accordingly (https://github.com/camunda/camunda-docs/pull/4187)

### Why should we do it?

Ensure users have a simpler out-of-the-box experience for [Camunda + connectors](https://github.com/camunda/product-hub/issues/2253#issuecomment-2277522319).

### PR DESCRIPTION
**Pull request by nikku**
### Proposed Changes

Implements https://github.com/camunda/camunda-modeler/issues/4455, specifically:

* [x] Adds a way to hook up with `activeTab.changed` in the app
* [x] Updates templates on switch to a Camunda 8 BPMN tab, tries every 24 hours
  * [x] Periodic checks are needed on MacOS where the app never really stops
* [x] `enable-connector-templates=true` is the default   
* [x] Change is test covered on the client
* [x] Change is test covered on the backend (app)
* [x] Updates what's new communication

Closes #4455
Related to https://github.com/camunda/camunda-docs/pull/4187

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] __Brief textual description__ of the changes present
* [ ] __Visual demo__ attached
* [ ] __Steps to try out__ present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

**Comment by nikku** [comment]
Proposal for messaging ("what's new communication"):

image

**Comment by nikku** [comment]
Another what's new proposal: 

image

**Comment by nikku** [comment]
What's new communication, v3:

image

File: client/src/plugins/version-info/ReleaseInfo.js [hunk 1]
```diff
@@ -44,8 +44,8 @@ export function ReleaseInfo(props) {
     <div className={ css.ReleaseInfo }>
       <ul className="dashed">
         <li>
-          <h4>Support for Camunda 8.6 features</h4>
-          This release adds support for upcoming Camunda 8.6 features, including execution listeners, the new binding type property, and converging inclusive gateways.
+          <h4>Camunda 8 Connector support</h4>
+          Use Camunda 8 Connectors to access a growing range of external services and communication protocols. In the modeler, you can now quickly and easily add pre-built Camunda connectors to your BPMN processes. To learn how to configure this feature, checkout the <a href="https://docs.camunda.io/docs/components/modeler/desktop-modeler/use-connectors/">connector documentation</a>.
```
Comment (barmac) [inline client/src/plugins/version-info/ReleaseInfo.js:48]: I'd drop "In the modeler" as this is the release info for the Modeler.

**Comment by barmac** [comment]
I've tested this and it works fine. There's only one issue on MacOS which was probably introduced with the feature (not this PR). When I close the window while the templates are being fetched, there's an unhandled error in the log:

```
(node:47611) UnhandledPromiseRejectionWarning: TypeError: Cannot read properties of null (reading 'webContents')
    at Object.send (/Users/maciej/workspace/bpmn-io/camunda-modeler/app/lib/util/renderer.js:68:18)
    at updateConnectorTemplates (/Users/maciej/workspace/bpmn-io/camunda-modeler/app/lib/connector-templates/index.js:71:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
(node:47611) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 4)
```

It's non-critical though.

File: client/src/plugins/version-info/ReleaseInfo.js (hunk 1 above)
Comment (nikku) [reply client/src/plugins/version-info/ReleaseInfo.js]: Adjusted via https://github.com/camunda/camunda-modeler/pull/4474/commits/29d2c130c657d5be698a6c2b16b1b72e1cb7b568.

**Comment by nikku** [comment]
> https://github.com/camunda/camunda-modeler/pull/4474#issuecomment-2307112666

Related to this I've seen the issue previously. We should be fail-safe and not fail when sending messages to non-existing main windows.

**Comment by nikku** [comment]
With https://github.com/camunda/camunda-modeler/pull/4474/commits/7dc67f4fdba9378522b1b2e6e11993ea3e4be371 we'll now log a warning to the console, and not attempt to send the message.

**Review by barmac — APPROVED** [review]
✅

**Comment by nikku** [comment]
CC @philippfromme as you worked on this previously.

File: app/lib/connector-templates/index.js [hunk 2]
```diff
@@ -176,4 +187,32 @@ function mergeConnectorTemplates(connectorTemplates, latestConnectorTemplates) {
...
\ No newline at end of file
+}
+
+const ONE_DAY_MS = 1000 * 60 * 60 * 24;
+
+/**
+ * @param {import('../util/renderer') } renderer
+ * @param {string} userPath
+ */
+function registerConnectorTemplateUpdater(renderer, userPath) {
```
Comment (philippfromme) [inline app/lib/connector-templates/index.js:198]: Nice! 👏🏻

**Review by philippfromme — COMMENTED** [review]
Good job! 💪🏻

**Pull request merged**